cli/azd/ci-build.ps1

```diff
@@ -5,6 +5,10 @@ param(
     [switch] $BuildRecordMode
 )
 
+# Enable the loopvar experiment, which makes the loop variaible for go loops like `range` behave as most folks would expect.
+# the go team is exploring making this default in the future, and we'd like to opt into the behavior now.
+$env:GOEXPERIMENT="loopvar"
```

Comment:
A minor nit: setting `$env:GOEXPERIMENT` here will persist the experiment in the user's shell session. This would only be a negative if the user switched to a different Go project and had some assumptions about the value of `$env:GOEXPERIMENT`. 

The fix here would be to record `$env:GOEXPERIMENT` in a temporary variable and reset its value at the end of the script, usually in a `finally` block so its execution is more assured.

Reply:
Learn something new about PowerShell every time I touch it!  Used the finally trick as you advised, @danieljurek!